src/main/java/de/uhd/ifi/se/decision/management/jira/persistence/ActiveObjectStrategy.java

```diff
@@ -255,9 +263,11 @@ public long insertLink(Link link, ApplicationUser user) {
 		return ACTIVE_OBJECTS.executeInTransaction(new TransactionCallback<Long>() {
 			@Override
 			public Long doInTransaction() {
-				for (LinkBetweenDifferentEntitiesEntity linkEntity : ACTIVE_OBJECTS.find(LinkBetweenDifferentEntitiesEntity.class)) {
-					if (linkEntity.getIdOfSourceElement().substring(1).equals(link.getSourceElement().getId()+"")
-							&& linkEntity.getIdOfDestinationElement().substring(1).equals(link.getDestinationElement().getId()+"")) {
+				for (LinkBetweenDifferentEntitiesEntity linkEntity : ACTIVE_OBJECTS
+						.find(LinkBetweenDifferentEntitiesEntity.class)) {
+					if (linkEntity.getIdOfSourceElement().substring(1).equals(link.getSourceElement().getId() + "")
```

Comment:
`getIdOfSourceElement().substring(1)` is a candidate for a well commented method, maybe named getAoIdOfSourceElement

Reply:
This method is now in the GenericLinkManager class. We need to get independet from the idWithPrefix.